File: app/src/main/java/com/uramonk/androidtemplateapp/component/WeatherComponent.java
package com.uramonk.androidtemplateapp.component;

import com.uramonk.androidtemplateapp.error.CommonErrorHandler;
import com.uramonk.androidtemplateapp.module.NetworkModule;
import com.uramonk.androidtemplateapp.module.WeatherModule;
import com.uramonk.androidtemplateapp.viewmodel.MainFragmentViewModel;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by uramonk on 2016/06/29.
 */
@Singleton
@Component(modules = {WeatherModule.class, NetworkModule.class})
public interface WeatherComponent {
    void inject(MainFragmentViewModel viewModel);
    void inject(CommonErrorHandler commonErrorHandler);
}

File: app/src/main/java/com/uramonk/androidtemplateapp/di/component/WeatherComponent.kt
package com.uramonk.androidtemplateapp.di.component

import com.uramonk.androidtemplateapp.di.module.NetworkModule
import com.uramonk.androidtemplateapp.di.module.StoreModule
import com.uramonk.androidtemplateapp.di.module.UseCaseModule
import com.uramonk.androidtemplateapp.di.module.WeatherModule
import com.uramonk.androidtemplateapp.error.CommonErrorHandler
import com.uramonk.androidtemplateapp.presentation.viewmodel.MainFragmentViewModel
import dagger.Component
import javax.inject.Singleton

/**
 * Created by uramonk on 2016/06/29.
 */
@Singleton
@Component(modules = arrayOf(WeatherModule::class, NetworkModule::class, UseCaseModule::class, StoreModule::class))
interface WeatherComponent {
    fun inject(viewModel: MainFragmentViewModel)
    fun inject(commonErrorHandler: CommonErrorHandler)
}
